lib/plugins/wallet/zcashd/zcashd.js

```diff
@@ -33,18 +33,14 @@ function checkCryptoCode (cryptoCode) {
 
 function accountBalance (cryptoCode) {
   return checkCryptoCode(cryptoCode)
-    .then(() => fetch('z_listaccounts'))
-    .then(accountsRes => _.isEmpty(accountsRes) ? fetch('z_getnewaccount') : Promise.resolve(_.first(accountsRes)))
-    .then(account => fetch('z_getbalanceforaccount', [account.account, 1]))
-    .then(res => new BN(res.pools.transparent.valueZat).plus(res.pools.sapling.valueZat).plus(res.pools.orchard.valueZat))
+    .then(() => fetch('getwalletinfo'))
+    .then(({ balance }) => new BN(balance).shiftedBy(unitScale).decimalPlaces(0))
```

Comment:
I don't know anything about Zcash, but aren't any of the other fields relevant? For example `shielded_balance`:

> the total confirmed shielded balance of the wallet in ZEC

https://zcash-rpc.github.io/getwalletinfo.html